src/xo-app/xoa-updates/index.js

```diff
@@ -0,0 +1,279 @@
+          <p key={key}>
+            <span className={textClasses[log.level]} >{log.date}</span>: <span dangerouslySetInnerHTML={{__html: ansiUp.ansi_to_html(log.message)}} />
+          </p>
+          ))}
+      </div>
+      <h2>Settings {configEdited ? '*' : ''}</h2>
+      <form className='form-inline'>
+        <fieldset>
+          <div className='form-group'>
+            <input
+              className='form-control'
+              placeholder='Host (myproxy.example.org)'
+              required
+              type='text'
+              value={configuration.proxyHost}
+              onChange={this._handleProxyHostChange}
+            />
+          </div>
+          {' '}
+          <div className='form-group'>
+            <input
+              className='form-control'
+              placeholder='Port (3128 ?...)'
+              required
+              type='text'
+              value={configuration.proxyPort}
+              onChange={this._handleProxyPortChange}
+            />
+          </div>
+          {' '}
+          <div className='form-group'>
+            <input
+              className='form-control'
+              placeholder='User name'
+              required
+              type='text'
+              value={configuration.proxyUser}
+              onChange={this._handleProxyUserChange}
+            />
+          </div>
+          {' '}
+          <div className='form-group'>
+            <Password
+              enableGenerator={false}
+              placeholder='password'
+              required
+              ref='proxyPassword'
+            />
+          </div>
+        </fieldset>
+        <br />
+        <fieldset>
+          <ActionButton type='submit' icon='save' btnStyle='primary' handler={this._configure}>Save</ActionButton>
+          {' '}
+          <button type='button' className='btn btn-default' onClick={this._handleConfigReset} disabled={!configEdited}>Reset</button>
+        </fieldset>
+      </form>
+      <h2>Registration</h2>
+      <p>
+        <strong>{registration.state}</strong>
+        {registration.email && <span> to {registration.email}</span>}
+        <span className='text-danger'>{registration.error}</span>
+      </p>
+      <form className='form-inline'>
+        <div className='form-group'>
+          <input
+            className='form-control'
+            placeholder='account email'
+            ref='email'
+            required
+            type='text'
+          />
+        </div>
+        {' '}
+        <div className='form-group'>
+          <Password
+            enableGenerator={false}
+            placeholder='password'
+            ref='password'
+            required
+          />
+        </div>
+        {' '}
+        <ActionButton type='submit' icon='success' btnStyle='primary' handler={this._register}>Register</ActionButton>
+      </form>
+      {/* FREE ONLY */}
+      <h2>Trial</h2>
+      {trial.state === 'default' && exposeTrial(trial.trial) &&
```

Comment:
If there are many cases, feel free to factorize in one or few dedicated functions (where you are able to use *early-`return`s*.